lib/Sema/TypeCheckType.cpp

```diff
@@ -3045,15 +3045,20 @@ Type TypeResolver::resolveTupleType(TupleTypeRepr *repr,
     complained = true;
   }
 
+  bool hadError = false;
   for (unsigned i = 0, end = repr->getNumElements(); i != end; ++i) {
     auto *tyR = repr->getElementType(i);
 
     Type ty = resolveType(tyR, elementOptions);
-    if (!ty || ty->hasError()) return ty;
+    if (!ty || ty->hasError())
```

Comment:
Wouldn't `return ErrorType::get(Context);` at this point have the same effect is at line 3059? I think possible reason why we didn't diagnose some of the problems in updated tests is related to this check returning empty type instead of an error.

Reply:
No. If we return in the middle of the for loop, then the QoI bug persists.

Reply:
Ah, because that problem is diagnosed by resolveType of course.